### PR TITLE
Add navigation bar component

### LINKED ISSUE #1: Add navigation bar component
- [ ] <b>Description</b>
    Add a navigation bar component.
- [ ] <b>Use cases</b>
    Navigation bar for a website.
- [ ] <b>Proposed design</b>
    Input parameters: website logo, width of navigation bar, colour, list of tab names
    Displays at the top of the screen and will be the full width.
- [ ] <b>Alternative design approaches</b>
    n/a
- [ ] <b>Workaround</b>
    n/a
- [ ] <b>Additional information</b>
    n/a

### PR DESCRIPTION
**Pull request by ngkelly**
## Description
Add a navigation bar component

Fixes #1.

## Author's Checklist
- [x] **These changes don't break existing dependants.** If they do, and you suggest to proceed, 
please explain why.

## Reviewer's Checklist
- [ ] **All classes have documentation comments.**
- [ ] **All methods have documentation comments.**
- [ ] **No parameters are hardcoded.** Should be loaded as properties instead. If there needs to be hardcoded properties, explain why.
- [ ] **All information to be logged/displayed to the user are internationalized.** If not, explain why. 
- [ ] **README.md has been updated if needed.**

## Assignee's Checklist
- [ ] **All GitHub action checks have passed.**
- [ ] **All reviewers have approved.**
- [ ] **Relevant documentation has been updated if needed.**

### CODE REVIEW

File: sample_app/src/App.js [hunk 1]
```diff
@@ -1,29 +1,36 @@
...
 /**
- * Sample React app with routes to components and a footer.
+ * Sample React app with a navigation bar, routes to components, and a footer.
  */
 class App extends React.Component {
 
+    constructor(props) {
+        super(props);
+        this.state = {
+            tabNames: ["Tab One", "Tab Two", "Tab Three", "Tab Four"],
```
Comment (the-c0d3br34k3r) [inline sample_app/src/App.js:17]: Would it be better to have a single JSON array instead of two arrays?

**Review by the-c0d3br34k3r — CHANGES_REQUESTED** [review]
Reviewed. Please complete the author's checklist as well.

File: sample_app/src/App.js (hunk 1 above)
Comment (ngkelly) [reply sample_app/src/App.js]: Yes, this has now been changed to a single JSON array.

File: README.md [hunk 2]
```diff
@@ -57,8 +57,13 @@ Describe the project.
 <!-- USAGE -->
 ## Usage
 
-### AppFooter
+### NavBar
+A navigation bar containing a logo and tabs.
+```
+<NavBar companyLogo={companyLogo} tabs={this.state.tabs}/>
```
Comment (the-c0d3br34k3r) [inline README.md:63]: How does `this.state.tabs` look like? Please show that as well.

**Review by the-c0d3br34k3r — CHANGES_REQUESTED** [review]
Requested changes to README.md.

File: README.md (hunk 2 above)
Comment (ngkelly) [reply README.md:92]: I have included it as well.

**Review by the-c0d3br34k3r — APPROVED** [review]

**Pull request merged**